### PR TITLE
Fix opcode global CallerApplicationID

### PR DESCRIPTION
**Pull request by thdailong**
Closes #x

## Proposed Changes

- Fix opcode global CallerApplicationID which previously, we have implemented it wrong way
## TODO

- [x] update change log

### CODE REVIEW

File: packages/runtime/src/ctx.ts
Comment (sczembor) [inline packages/runtime/src/ctx.ts:218]: lets add more description

**Review by sczembor — COMMENTED** [review]
there is one test failing. Please take a look at the comments I left

File: CHANGELOG.md
Comment (PabloLION) [reply CHANGELOG.md:65]: It gives the current application ID. See https://github.com/scale-it/algo-builder/pull/798

File: packages/runtime/src/ctx.ts
Comment (PabloLION) [inline packages/runtime/src/ctx.ts:219]: It seems you are using my code from https://github.com/scale-it/algo-builder/pull/798/files but not with my commit nor adding me as a co-author. This is not very fair.

File: packages/runtime/src/ctx.ts
Comment (PabloLION) [inline packages/runtime/src/ctx.ts:220]: Good job on this new PR. Though it seems that you are using my commit from https://github.com/scale-it/algo-builder/pull/798/files but not adding commit in that branch, nor adding me as a co-author. TBH it feels not very fair to me.

Comment (PabloLION) [reply packages/runtime/src/ctx.ts:220]: Actually, I think my fix was also not so perfect, tho it gives the wanted result. 
And it looks correct to me what you said here in https://github.com/scale-it/algo-builder/pull/798#issuecomment-1268772011.

File: packages/runtime/test/src/interpreter/opcode-list.ts
Comment (PabloLION) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3057]: The problem is that the current application id is not pushed to the `innerTxAppIDCallStack`. Getting the second last element makes more sense when `currentAppId` was pushed to the call stack, or otherwise we should still use the last one and remove the action of `push`.
Here, the last element of `interpreter.runtime.ctx.innerTxAppIDCallStack = [1, 2];` was not coherent with `CurrentAppID`. Something is still wrong. (tho tests all pass)

**Review by PabloLION — COMMENTED** [review]
Seems to me that more changes are needed...

File: packages/runtime/test/src/interpreter/opcode-list.ts
Comment (thdailong) [reply packages/runtime/test/src/interpreter/opcode-list.ts:3057]: Actually this one is for unit test. Since you see that, we set up the inner transaction application call ID in [here](https://github.com/scale-it/algo-builder/pull/809/files#diff-0bd11169d40e3be81fef88fe8a8e4cf0795255b19d1f4d44d615ea872a1e0e90R3039). So the comment here https://github.com/scale-it/algo-builder/pull/798#issuecomment-1268772011 is where I misunderstand you

File: CHANGELOG.md
Comment (thdailong) [reply CHANGELOG.md:65]: Let me update the changelog

File: packages/runtime/src/ctx.ts
Comment (thdailong) [reply packages/runtime/src/ctx.ts:220]: Oh sorry for making this unfair to you! I am not really familiar with git but I will make add your commit to this branch and make you a co-author

Comment (PabloLION) [reply packages/runtime/src/ctx.ts:221]: It's ok 😃 I already have the "contributor" chip. You can do it the way you like!

File: packages/runtime/test/src/interpreter/opcode-list.ts
Comment (PabloLION) [reply packages/runtime/test/src/interpreter/opcode-list.ts:3057]: Oh, ok. That's what I was thinking. Then your fix is what I was trying (but failed) to do. Congrats on finishing it 👍

Comment (PabloLION) [reply packages/runtime/test/src/interpreter/opcode-list.ts:3057]: I'm not the creator not have write access in this repo. So you'd have to resolve these conversations by yourself @thdailong

File: packages/runtime/src/ctx.ts
Comment (sczembor) [inline packages/runtime/src/ctx.ts:219]:
```suggestion
		// when the top value in the `innerTxAppIDCallStack` is the own appID
		// not the caller appID(we need to get the second top one)
```

**Review by sczembor — APPROVED** [review]
Good job, please check the comment before merging

**Pull request merged**
